src/numbers_royale.cpp

```diff
@@ -0,0 +1,185 @@
+#include "numbers_royale.h"
+#include "cpu.h"
+
+
+// Helper functions
+/**
+ * @brief Wait for user to press enter
+ * 
+ * @param message Message to display
+ */
+void wait_for_enter(const char* message = "Press <enter> to continue... ") {
+  printf("%s", message);
+  std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+}
+
+/**
+ * @brief Get the validated input object
+ * 
+ * @param is_valid lambda function to validate input
+ * @return int selected input after validation
+ */
+int get_validated_input(const std::function<bool(int)>& is_valid) {
+  int input;
+  while (true) {
+    std::cin >> input;
+    if (std::cin.fail() || !is_valid(input)) {
+      std::cin.clear(); // clear the error flag
+      std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // discard invalid input
+      std::cout << "Invalid input. Please try again: ";
+    } else {
+      std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // discard any extra input
+      return input;
+    }
+  }
+}
+
+
+// Numbers Royale Class
+NumbersRoyale::NumbersRoyale(GUI *gui): gui_(gui), board_(gui) {
+  gui_->clear_screen();
+}
+
+void NumbersRoyale::main_menu() {
+  while (true) {
+    gui_->print_header("NUMBERS ROYALE");
+    gui_->print_item("1. Play Game");
+    gui_->print_item("2. How to Play");
+    gui_->print_item("3. Quit");
+    gui_->print_border();
+    printf("Select option: ");
+    int option = get_validated_input([](int input) { 
+      return input >= MainMenuOption::PLAY_GAME && input <= MainMenuOption::QUIT; 
+    });
+
+    switch (option) {
+      case MainMenuOption::PLAY_GAME:
+        play_mode();
+        break;
+      case MainMenuOption::HOW_TO_PLAY:
+        how_to_play();
+        break;
+      case MainMenuOption::QUIT:
+        return;
+      default:
+        std::cerr << "Error: Invalid option" << std::endl;
+        break;
+    }
+  }
+}
+
+void NumbersRoyale::how_to_play() {
+  // TODO: Implement line wrapping for gui.h
+  gui_->clear_screen();
+  printf("|=================================|\n");
+  printf("|           HOW TO PLAY           |\n");
+  printf("|=================================|\n");
+  printf("| 1. The game is played by two    |\n");
+  printf("|    players.                     |\n");
+  printf("| 2. Each player will take turns  |\n");
+  printf("|    selecting a number from      |\n");
+  printf("|    their available numbers.     |\n");
+  printf("| 3. The player who selects the   |\n");
+  printf("|    larger number wins the round.|\n");
+  printf("| 4. After using a number, it     |\n");
+  printf("|    cannot be used again.        |\n");
+  printf("| 5. The game ends when all the   |\n");
+  printf("|    numbers have been selected.  |\n");
+  printf("| 6. The player with the most     |\n");
+  printf("|    points wins the game.        |\n");
+  printf("|=================================|\n");
```

Comment:
this should be in GUI imo

Reply:
Yeah I got too lazy to implement line wrapping by word in the GUI so this is just hard coded :)